kroxylicious/src/main/java/io/kroxylicious/proxy/internal/net/EndpointRegistry.java

```diff
@@ -515,43 +536,53 @@ public CompletionStage<VirtualClusterBinding> resolve(Endpoint endpoint, String
         return lcr.bindingStage().thenApply(acceptorChannel -> {
             var bindings = acceptorChannel.attr(CHANNEL_BINDINGS);
             if (bindings == null || bindings.get() == null) {
-                throw buildEndpointResolutionException("No channel bindings found for", endpoint, sniHostname);
+                throw buildEndpointResolutionException(NO_CHANNEL_BINDGINS_MESSAGE, endpoint, sniHostname);
             }
             // We first look for a binding matching by SNI name, then fallback to a null match.
             var binding = bindings.get().getOrDefault(RoutingKey.createBindingKey(sniHostname), bindings.get().get(RoutingKey.NULL_ROUTING_KEY));
             if (binding == null) {
                 // If there is an SNI name that matches against the virtual cluster broker address pattern, we generate
                 // a restricted broker binding that points at the virtual cluster's bootstrap.
                 if (sniHostname != null) {
-                    var allBindingsForPort = bindings.get().values();
-                    var brokerAddress = new HostPort(sniHostname, endpoint.port());
-                    var allBootstrapBindings = getAllBootstrapBindings(allBindingsForPort);
-                    var bootstrapToBrokerId = allBootstrapBindings.stream()
-                            .collect(HashMap<VirtualClusterBootstrapBinding, Integer>::new, (m, b) -> {
-                                var nodeId = b.virtualCluster().getBrokerIdFromBrokerAddress(brokerAddress);
-                                if (nodeId != null) {
-                                    m.put(b, nodeId);
-                                }
-                            }, HashMap::putAll);
+                    Map<VirtualClusterBootstrapBinding, Integer> bootstrapToBrokerId = mapSniHostnameToBindings(endpoint, sniHostname, bindings);
```

Comment:
wondering if 
```suggestion
                    Map<VirtualClusterBootstrapBinding, Integer> bootstrapToBrokerId = mapSniHostnameToBootstrapBinding(endpoint, sniHostname, bindings);
```
Is a better name? Or even just 
```suggestion
                    Map<VirtualClusterBootstrapBinding, Integer> bootstrapToBrokerId = bootstrapBindings(endpoint, sniHostname, bindings);
```

Reply:
Wondering if it should have a word in there to show that it may not find any bindings, like `findBootstrapBindings(..)`?

Reply:
`findBootstrapBindings` is the best yet.

Reply:
fixed in 00263c001433962a1f38e0e3ca4de32931c1c7d3